test/test_crc32.cc

```diff
@@ -195,6 +197,20 @@ class crc32_variant : public ::testing::TestWithParam<crc32_test> {
     }
 };
 
+/* Specifically to test where we had dodgy alignment in the acle CRC32
+ * function. All others are either byte level access or use intrinsics
+ * that work with unaligned access */
+class crc32_align : public ::testing::TestWithParam<int> {
+public:
+    void hash(int param, crc32_func crc32) {
+        uint8_t *buf = (uint8_t*)zng_alloc(sizeof(uint8_t) * (128 + param));
+        if (buf != NULL) {
+            (void)crc32(0, buf + param, 128);
+        }
+        zng_free(buf);
+    }
+};
```

Comment:
Our alignment offset won't be negative because we control the test parameters and it would never make sense to do so, as all non zero modulus values of 8 are tested.